src/main.rs

```diff
@@ -31,6 +31,7 @@ struct Cli {
     lib: bool,
 }
 
+/// This type represents the different formats of dependencies given by the user
```

Comment:
```suggestion
```

You don't need doc comment on private types. You can if you think your comment provides useful extra information but in this particular case you're just saying "the type `Dependency` represents a dependency given by the user" XD

Reply:
Its like extra signs for help reader who don't speak the rustacean's language. I can change that into comments instead

Reply:
It's not rustacean language. A dependency is a dependency. If you create a class called `Dependency` in Python, the developer should assume that this class represents a dependency. Same for Java, JS, etc...

Saying that a type represents something is also not a helpful information because data are supposed to represent something.

What I'm getting at is that every comment you add needs to bring value. Stating the obvious is only making the code heavier to read.

Reply:
Oh there is a recent (7 days ago) lengthy post on the orange site about that (also shared on reddit). Someone made a blog post defending the opposite of what I'm teaching you :grin:

Reply:
Other than that, here is a post I found very good :thinking: https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/